plugins/modules/lvm_pv.py

```diff
@@ -0,0 +1,205 @@
+            return False
+    else:
+        module.warn("Rescan path not found for device {0}".format(device))
+        return False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            device=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            force=dict(type='bool', default=False),
+            resize=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+    )
+
+    device = module.params['device']
+    state = module.params['state']
+    force = module.params['force']
+    resize = module.params['resize']
+    changed = False
+    actions = []
+
+    # Validate device existence for present state
+    if state == 'present' and not os.path.exists(device):
+        module.fail_json(msg="Device %s not found" % device)
+
+    is_pv = get_pv_status(module, device)
+
+    if state == 'present':
+        # Create PV if needed
+        if not is_pv:
+            if module.check_mode:
+                changed = True
+                actions.append('created')
+            else:
+                cmd = ['pvcreate']
+                if force:
+                    cmd.append('-f')
+                cmd.append(device)
+                rc, out, err = module.run_command(cmd)
+                if rc != 0:
+                    module.fail_json(msg="Failed to create PV: %s" % err)
+                changed = True
+                actions.append('created')
+            is_pv = True
+
+        # Handle resizing
+        if resize and is_pv:
+            if not module.check_mode:
+                # Perform device rescan if each time
+                if rescan_device(module, device):
+                    actions.append('rescanned')
+                original_size = get_pv_size(module, device)
+                rc, out, err = module.run_command(['pvresize', device])
+                if rc != 0:
+                    module.fail_json(msg="PV resize failed: %s" % err)
+                new_size = get_pv_size(module, device)
+                if new_size != original_size:
+                    changed = True
+                    actions.append('resized')
+            else:
+                # In check mode, assume resize would change
+                changed = True
+                actions.append('would be resized')
+
+    elif state == 'absent':
+        if is_pv:
+            if module.check_mode:
+                changed = True
+                actions.append('would be removed')
+            else:
+                cmd = ['pvremove']
+                if force:
+                    cmd.extend(['-ff', '-y'])
+                else:
+                    cmd.append('-y')
+                cmd.append(device)
+                rc, out, err = module.run_command(cmd)
+                if rc != 0:
+                    module.fail_json(msg="Failed to remove PV: %s" % err)
```

Comment:
This _could_ be made simpler with:
```suggestion
                rc, out, err = module.run_command(cmd, check_rc=True)
```
Potentially applies to all other commands here.

Reply:
Applied